src/sortStudents.ts

```diff
@@ -1,16 +1,52 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade'
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function averageGrade({ grades }: Student): number {
+  return grades.reduce((acc, grade) => acc + grade, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student[],
+  sortBy: SortType,
+  order: SortOrder,
+): Student[] {
+  return [...students].sort((prevStudent, nextStudent) => {
+    switch (sortBy) {
+      case SortType.Name:
+      case SortType.Surname:
+        return order === 'asc'
+          ? prevStudent[sortBy].localeCompare(nextStudent[sortBy])
+          : nextStudent[sortBy].localeCompare(prevStudent[sortBy]);
+
+      case SortType.Age:
+      case SortType.Married:
+        return order === 'asc'
+          ? Number(prevStudent[sortBy]) - Number(nextStudent[sortBy])
+          : Number(nextStudent[sortBy]) - Number(prevStudent[sortBy]);
+
+      case SortType.AverageGrade:
+        return order === 'asc'
+          ? averageGrade(prevStudent) - averageGrade(nextStudent)
+          : averageGrade(nextStudent) - averageGrade(prevStudent);
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+      default:
+        throw new Error('Unknown Sort type');
```

Comment:
```suggestion
        throw new Error(`Unknown Sort type: ${sortBy}`);
```
Make debugging easier